src/java.base/share/classes/java/lang/package-info.java

```diff
@@ -25,35 +25,42 @@
 
 /**
  * Provides classes that are fundamental to the design of the Java
- * programming language. The most important classes are {@code
- * Object}, which is the root of the class hierarchy, and {@code
+ * programming language. The most important classes are {@link
+ * Object}, which is the root of the class hierarchy, and {@link
  * Class}, instances of which represent classes at run time.
  *
  * <p>Frequently it is necessary to represent a value of primitive
- * type as if it were an object. The wrapper classes {@code Boolean},
- * {@code Character}, {@code Integer}, {@code Long}, {@code Float},
- * and {@code Double} serve this purpose.  An object of type {@code
- * Double}, for example, contains a field whose type is double,
- * representing that value in such a way that a reference to it can be
- * stored in a variable of reference type.  These classes also provide
- * a number of methods for converting among primitive values, as well
- * as supporting such standard methods as equals and hashCode.  The
- * {@code Void} class is a non-instantiable class that holds a
- * reference to a {@code Class} object representing the type void.
+ * type as if it were an object.<a id="wrapperClasses"></a> The
+ * <dfn>{@index "wrapper classes"}</dfn> {@link Boolean}, {@link
```

Comment:
I can see that JDK uses the `<a name="xyz"></a>` pattern in many locations. That is, an empty `a` tag which has the `name` attribute. You use the `id` attribute. JDK uses such pattern too, but only in a few locations.

Regardless of the details, I was wondering if instead of adding a new tag with an attribute, we should just add an attribute to an existing tag.

`dfn` seems to be designed exactly for that [^1][^2]. Additionally, keeping tags to a minimum might not worsen readability as much:

    <dfn id="wrapperClasses">{@index "wrapper classes"}</dfn>

@jonathan-gibbons, @hns, thoughts?

[^1]: https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-dfn-element
[^2]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dfn

Reply:
Agree; I have recommedned consolidating id property into h2 without an extra a tag.

Reply:
`<dfn>` is a semantic tag to indicate the defining instance of a term. It may be used by search engines, to improve their results.   When `<dfn>` is used as intended, it may be reasonable and convenient to put an `id` on the tag, to provide a link target for elsewhere in the documentation. It may also be convenient to add `id` to other locations, especially headers, but note that `javadoc` now does that automatically.

The usage of `<a name="...">` is a legacy usage from HTML 4, before the improved rules for `id` in HTML 5.  It would be a reasonable cleanup to move away from such tags, putting an equivalent `id` on either a replacement tag (such as `<span>`) or on an appropriate nearby tag.

Reply:
Thanks for the HTML tip; will add an id to the dfn tag.

Reply:
Fair enough, @hns. Initially, I thought to reply to you that we don't seem to specify the form of id generated by `@index` and that, in fact, it would be more error-prone for an author to guess/compute it in their head. For example, I would expect automatically generated ids to be camelCased.

But then I recalled the design of the `@index` tag and realised that the author does not have to think about the id. Just like index in good old paper books, javadoc index  does not discriminate between referring and the defining sites of a term. Sites are simply grouped by the term.

However, if the term is even slightly different ("wrapper class" vs "wrapper classES"), the author does not get the desired indexing without working around javadoc. Seems like `@index` might be too inflexible.

Here's a real case of bad index. Those entries are really about the same term, but that's currently inexpressible with `@index` without restructuring one of the sites:

* https://docs.oracle.com/en/java/javase/23/docs/api/java.compiler/javax/lang/model/package-summary.html#Javalanguagemodel

* https://docs.oracle.com/en/java/javase/23/docs/api/java.compiler/module-summary.html#LanguageModel

Reply:
@pavelrappo if the goal is to define a different search term/anchor, then an explicit additional `id` is certainly justified. Regarding the last example, it is ok to use the same search term on different pages/classes. It's only within one HTML page that duplicate id's are invalid.

Reply:
I'm going to push this PR as-is. Any cleanup in index terms and `id`s can happen in a subsequent changeset. Thanks.